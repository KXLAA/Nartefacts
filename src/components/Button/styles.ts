import styled from 'styled-components'
import { ButtonProps } from '@/components/Button'
import { colors } from '@/styles/global'

const { grayPrimary, blackSecondary } = colors

type StyledButtonProps = Pick<
  ButtonProps,
  'buttonColor' | 'textColor' | 'fullWidth' | 'buttonType'
>

export const Button = styled.button<StyledButtonProps>`
  display: flex;
  max-width: ${({ fullWidth }) => (fullWidth ? '100%' : '301px')};
  width: 100%;
  height: 82px;
  align-items: center;
  justify-content: center;
  padding: 16px 20px;
  background: ${({ buttonColor }) => (buttonColor ? buttonColor : 'none')};
  font-size: 36px;
  border: ${({ buttonType, textColor }) =>
    buttonType === 'link-outline' || buttonType === 'outline'
      ? `solid 4px ${textColor}`
      : 'none'};
  border-radius: 8px;
  color: ${({ textColor }) => textColor};
  font-weight: bold;
  cursor: pointer;
  transition: all 0.3s ease;
  &:hover {
    background-color: #dddddd;
  }

  &:active {
    transform: translateX(0rem) translateY(0.125rem);
  }

  &:disabled,
  &:disabled:hover,
  &:disabled:active {
    background-color: ${grayPrimary};
    color: ${blackSecondary};
    cursor: not-allowed;
    transform: none;
  }
`

export const Link = styled.a<StyledButtonProps>`
  display: flex;
  max-width: 301px;
  width: 100%;
  height: 82px;
  align-items: center;
  justify-content: center;
  padding: 16px 20px;
  background-color: ${({ buttonColor }) => buttonColor || grayPrimary};
  font-size: 36px;
  border: none;
  border-radius: 8px;
  color: ${({ textColor }) => textColor || blackSecondary};
  font-weight: bold;
  cursor: pointer;
  transition: all 0.3s ease;
  &:hover {
    background-color: #dddddd;
  }

  &:active {
    transform: translateX(0rem) translateY(0.125rem);
  }
`
