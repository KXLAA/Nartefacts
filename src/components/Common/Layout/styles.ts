/* istanbul ignore file */

import { styled } from '@/lib/stitches.config'

export const StyledLayout = styled('main', {
  width: '100%',
  margin: '0 auto',

  variants: {
    size: {
      sm: {
        maxWidth: '400px',
        py: '$10',
        px: '$8',
      },
      md: {
        maxWidth: '800px',
        py: '$10',
        px: '$8',
      },
      lg: {
        maxWidth: '1280px',
        py: '$10',
        px: '$8',
      },
    },
  },

  defaultVariants: {
    size: 'lg',
  },
})

export const Grid = styled('div', {
  boxSizing: 'border-box',
  display: 'grid',
  width: '100%',

  variants: {
    align: {
      start: {
        alignItems: 'start',
      },
      center: {
        alignItems: 'center',
      },
      end: {
        alignItems: 'end',
      },
      stretch: {
        alignItems: 'stretch',
      },
      baseline: {
        alignItems: 'baseline',
      },
    },
    justify: {
      start: {
        justifyContent: 'start',
      },
      center: {
        justifyContent: 'center',
      },
      end: {
        justifyContent: 'end',
      },
      between: {
        justifyContent: 'space-between',
      },
    },
    flow: {
      row: {
        gridAutoFlow: 'row',
      },
      column: {
        gridAutoFlow: 'column',
      },
      dense: {
        gridAutoFlow: 'dense',
      },
      rowDense: {
        gridAutoFlow: 'row dense',
      },
      columnDense: {
        gridAutoFlow: 'column dense',
      },
    },
    columns: {
      1: {
        gridTemplateColumns: 'repeat(1, 1fr)',
      },
      2: {
        gridTemplateColumns: 'repeat(2, 1fr)',
      },
      3: {
        gridTemplateColumns: 'repeat(3, 1fr)',
      },
      4: {
        gridTemplateColumns: 'repeat(4, 1fr)',
      },
    },
    gap: {
      1: {
        gap: '$1',
      },
      2: {
        gap: '$2',
      },
      3: {
        gap: '$3',
      },
      4: {
        gap: '$4',
      },
      5: {
        gap: '$5',
      },
      6: {
        gap: '$6',
      },
      7: {
        gap: '$8',
      },
      8: {
        gap: '$10',
      },
      9: {
        gap: '$12',
      },
    },
    gapX: {
      1: {
        columnGap: '$1',
      },
      2: {
        columnGap: '$2',
      },
      3: {
        columnGap: '$3',
      },
      4: {
        columnGap: '$4',
      },
      5: {
        columnGap: '$5',
      },
      6: {
        columnGap: '$6',
      },
      7: {
        columnGap: '$8',
      },
      8: {
        columnGap: '$10',
      },
      9: {
        columnGap: '$12',
      },
    },
    gapY: {
      1: {
        rowGap: '$1',
      },
      2: {
        rowGap: '$2',
      },
      3: {
        rowGap: '$3',
      },
      4: {
        rowGap: '$4',
      },
      5: {
        rowGap: '$5',
      },
      6: {
        rowGap: '$6',
      },
      7: {
        rowGap: '$8',
      },
      8: {
        rowGap: '$10',
      },
      9: {
        rowGap: '$12',
      },
    },
  },
  defaultVariants: {
    gap: 4,
  },
})
