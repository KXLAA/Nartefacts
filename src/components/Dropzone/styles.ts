import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Drop = styled.div`
  background: #202020;
  border: 1px dashed #363636;
  box-sizing: border-box;
  border-radius: 8px;
  width: 100%;
  max-width: 800px;
  height: 800px;
  padding: 24px;
  cursor: pointer;
  transition: all 0.3s ease;
  &:hover {
    background: #d8d8d8;
    transform: translateX(0rem) translateY(0.125rem);
  }
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-height: 100%;
  gap: 24px;
  p {
    font-size: 24px;
    font-weight: bold;
    width: 70%;
    text-align: center;
    color: #5a5a5a;
  }
`
