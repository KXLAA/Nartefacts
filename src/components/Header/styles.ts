import styled from 'styled-components'

export const Layout = styled.header`
  display: flex;
  flex-direction: column;
  gap: 3rem;
  width: 100%;
`

export const Nav = styled.nav`
  display: flex;
  justify-content: space-between;
  align-items: center;
  justify-items: center;
`

export const Spacer = styled.div`
  width: 48px;
  height: 48px;
`
