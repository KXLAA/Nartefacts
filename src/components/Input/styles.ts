import styled from 'styled-components'

export const Wrapper = styled.div``

export const Input = styled.input`
  width: 100%;
  height: 80px;
  background: none;
  border: 1px solid #5a5a5a;
  border-radius: 8px;
  padding: 1rem;
  color: #5a5a5a;
  font-size: 1.5rem;
`
