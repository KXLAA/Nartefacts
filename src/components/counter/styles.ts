import { styled } from '@/lib/stitches'

export const StyledWrapper = styled('div', {
  textAlign: 'center',
  display: 'flex',
  justifyContent: 'center',
})

export const StyledCounter = styled('div', {
  backgroundColor: '$black-light',
  fontWeight: 'bold',
  width: '100%',
  borderRadius: '$rounded',
  maxWidth: '500px',
  fontSize: '$lg',
  py: '$1',
})
