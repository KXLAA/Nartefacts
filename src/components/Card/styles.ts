import styled from 'styled-components'

import { getBorder, getSize } from '@/styles/global'

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${getSize('lg')};
`

export const Gradient = styled.div`
  background: linear-gradient(
    to bottom,
    rgba(0, 0, 0, 0.5),
    rgba(0, 0, 0, 0.5)
  );
  width: 100%;
  height: 122px;
  border: ${getBorder('md', 'solid', 'grayBase')};
  border-radius: ${getSize('xs')};
`

export const ImageWrapper = styled.div`
  border: ${getBorder('md', 'solid', 'grayBase')};
  border-radius: ${getSize('xs')};
  span {
    border-radius: ${getSize('xs')};
  }
`
