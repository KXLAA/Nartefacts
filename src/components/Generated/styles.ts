import styled from 'styled-components'

import { getSize, getColor } from '@/styles/global'

export const ImageWrapper = styled.div<{ small?: boolean }>`
  border: ${({ small }) => (small ? '1px' : getSize('xxs'))} solid
    ${getColor('grayLight')};
  border-radius: ${({ small }) => (small ? getSize('xs') : getSize('sm'))};
  padding: 0;

  span {
    border-radius: ${getSize('sm')};
  }
`
