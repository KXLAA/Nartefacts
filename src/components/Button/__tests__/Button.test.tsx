import { render } from '@testing-library/react'
import { colors } from 'styles/global'
const { grayPrimary, blackSecondary } = colors
import { RouterContext } from 'next/dist/shared/lib/router-context'
import { createMockRouter } from 'utils'
import { Button } from '..'

describe('<Button />', () => {
  it('should render the button with default values & styles', () => {
    const { getByRole } = render(<Button />)
    const button = getByRole('button', { name: /Button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': grayPrimary,
      color: blackSecondary,
    })
  })

  it('should render the Button correctly', () => {
    const { container } = render(<Button />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the button with correct values based on props', () => {
    const { getByRole } = render(
      <Button buttonType="secondary" text="Props" fullWidth />,
    )
    const button = getByRole('button', { name: /props/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': blackSecondary,
      color: grayPrimary,
      'max-width': '100%',
    })
  })

  it('Button Link should render as an <a> tag with href attribute', () => {
    const { getByRole } = render(
      <RouterContext.Provider
        value={createMockRouter({ query: { id: '33' }, pathname: 'kola' })}
      >
        <Button buttonType="link-primary" href="/test" />
      </RouterContext.Provider>,
    )

    const buttonLink = getByRole('link', { name: /button/i })
    expect(buttonLink).toBeInTheDocument()
    expect(buttonLink).toHaveAttribute('href', '/test')
  })
})
