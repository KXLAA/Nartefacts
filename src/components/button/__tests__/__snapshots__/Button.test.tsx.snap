// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render <Button /> correctly with the default styles 1`] = `
<button
  class="nartefacts-c-hctSdv nartefacts-c-hctSdv-gHFwzb-size-lg nartefacts-c-hctSdv-eDnYwL-variant-light nartefacts-c-hctSdv-deuXAd-fullWidth-true nartefacts-c-hctSdv-iPJLV-css"
>
  Button CTA
</button>
`;
